receiver/hostmetricsreceiver/internal/scraper/memoryscraper/memory_scraper_test.go

```diff
@@ -35,27 +36,45 @@ func TestScrape(t *testing.T) {
 		name              string
 		virtualMemoryFunc func() (*mem.VirtualMemoryStat, error)
 		expectedErr       string
+		initializationErr string
+		config            *Config
+		bootTimeFunc      func() (uint64, error)
```

Comment:
Looks like there is no test case using `bootTimeFunc`. We should either add the test case or remove `bootTimeFunc` along with `bootTime` scraper field

Reply:
@abhide please take a look at this comment as well

Reply:
`bootTime` is used here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/7312/files#diff-817b256eb274e3c300f40f2eeeede0f69acf56f06d9cd5597b86831a65c62017R53 

Added a test for `bootTimeFunc`

Reply:
It can be removed from [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/7312/files#diff-817b256eb274e3c300f40f2eeeede0f69acf56f06d9cd5597b86831a65c62017R53) if it's not being used in other places. bootTime can found right in `start` function from `host.BootTime`, no need to pass it through the struct. Using it in tests is the only reason to keep it

Reply:
Yeah; we can keep it. I added a unit tests that checks for initializationError.